crates/bevy_animation/src/graph.rs

```diff
@@ -0,0 +1,109 @@
+//! The animation graph.
+
+use std::ops::{Index, IndexMut};
+
+use bevy_asset::{Asset, Handle};
+use bevy_reflect::Reflect;
+use petgraph::graph::{DiGraph, NodeIndex};
+use petgraph::visit::{Dfs, Visitable};
+use petgraph::Graph;
+
+use crate::AnimationClip;
+
+#[derive(Asset, Reflect, Debug)]
+pub struct AnimationGraph {
+    #[reflect(ignore)]
```

Comment:
This is going to be a problem all the way, no? How do we plan to serialize that asset if `petgraph` has no serializable type?

Reply:
Or is the plan to derive `Reflect` manually eventually?

Reply:
This is now fixed. It took a few tries to figure out the right design. I'm going through petgraph's native serialization.